### DISCUSSION TITLE
Does this work?: Audio NumPy Array » ffmpegio audio-processing » NumPy Array

### DISCUSSION THREAD

**Discussion by schittli** [python-ffmpegio/python-ffmpegio#13] [Q&A] [answered]
Good afternoon

**thank you very much for sharing your great work!**, it's very useful to have `ffmpegio`.

Because ffmpeg has great audio-/video-processing features, I had this idea:

#### Is it possible to use `ffmpegio` to process numpy arrays with ffmpeg audio/video processors without having to create files on the disc?

I have a practical, simple example:
1. Stereo audio data is stored in a numpy array
2. The numpy array is passed to `ffmpegio`
3. `ffmpegio` uses ffmpeg and converts the stereo signal to mono
4. Afterwards, the result is passed back to Python as numpy Array

_(I use this example because if one mathematically convert stereo to mono, the result is about 3dB less loud than the stereo signal and there can be spikes with large peaks. So one must start a postprocessor after the mono conversion - what ffmpeg does very well 😃)_

Thanks a lot for any help!, kind regards,
Thomas

**Comment by tikuma-lsuhsc** [comment, marked as answer]
@schittli - That's an interesting usecase that I didn't think of. I had to do some digging to find a workaround, and luckily there is an `anull` filter, which does nothing but just pass the input samples to output. With it, you can use the `filter` function.

Here is a quick example:

```python
import ffmpegio as ff
import numpy as np
from matplotlib import pyplot as plt

fs = 16000
T = 1
x = np.random.default_rng().integers(2^-15,2^15,dtype='int16',size=(T*fs,2))

fsout,y = ff.audio.filter('anull', fs, x, ac=1) # no explicit filtering, just implicit stereo->mono conversion

plt.plot(x)
plt.plot(y)
plt.show()
```

Let me know if this works for you

**Reply by schittli** [reply]
Good evening @tikuma-lsuhsc 

**thank you very, very much for your help**, your solution perfectly works.

I am thrilled how elegantly you solved this task… it is ingenious that we can use ffmpeg via `ffmpegio` so easily and transparently without having to work with temporary files.

Thank you very much and all the best!,
kind regards, Thomas

**Reply by tikuma-lsuhsc** [reply]
Glad you find it useful. Feel free to drop in for more questions! -Kesh

**Reply by SohamTilekar** [reply]
I read the Conversation Dose that means the FFMPEGIO do not support the Stereo audio to Stereo audio NumPy array.

**Reply by tikuma-lsuhsc** [reply]
It should handle any channel configuration that FFmpeg can handle. It's his specific usecase in which he wanted a mono output.
